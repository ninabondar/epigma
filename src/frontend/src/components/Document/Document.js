import React from "react"
import BEM from "../../utils/BEM"
import { formatDistanceToNow } from "date-fns"

import "./Document.scss"
const b = BEM("Document")

const Document = ({ createdAt, title, clickHandler }) => (
  <div className={b()}>
    <div className={b("title")}>{title}</div>
    <div className={b("side-info")}>
      <small>{formatDistanceToNow(createdAt)} ago</small>
      <div className={b("more-btn")} onClick={clickHandler} />
    </div>
  </div>
)

Document.defaultProps = {
  clickHandler: () => {}
}

export default Document
