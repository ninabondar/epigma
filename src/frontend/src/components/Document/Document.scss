@import "../../styles/variables";

.Document {
  box-sizing: border-box;
  background: #ffffff;
  border-radius: 2px;
  padding: 20px 16px;
  width: calc(100vw / 5 - (3 * 0.5em));
  height: 88px;
  float: left;
  margin: 0.5em;
  position: relative;

  &__title {
    font-size: $FONT_SIZE_GENERAL;
    white-space: nowrap;
    text-overflow: ellipsis;
    width: 100%;
    display: block;
    overflow: hidden;
  }

  &__side-info {
    box-sizing: border-box;
    padding-top: 10px;
    font-size: $SMALL_FONT_SIZE;
    color: $PALE_GRAY;
    line-height: $SMALL_LINE_HEIGHT;
    display: flex;
    position: relative;
    justify-content: space-between;
  }

  &__more-btn {
    width: 20px;
    border: 0;
    outline: none;
    background-color: transparent;
    background-position: right;
    background-size: 60%;
    background-repeat: no-repeat;
    background-image: url("../../images/dustbin.svg");
    cursor: pointer;
    opacity: 0.6;
    box-sizing: border-box;

    &:hover {
      background-image: url("../../images/dustbin_black.svg");
      opacity: 1;
      & ~ .Document__more {
        display: block;
      }
    }
  }

  &__more {
    display: none;
    min-width: 70px;
    position: absolute;
    right: 4px;
    top: -80%;
    color: white;
    border-radius: 2px;
    box-sizing: border-box;
    padding: 6px 0;
    background: $HEADER_PANEL_COLOR;
  }
}

@media screen and (max-width: 820px) {
  .Document {
    width: 95%;
  }
}