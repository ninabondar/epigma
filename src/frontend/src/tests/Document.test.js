import React from 'react'
import Document from '../components/Document'

describe('<Document/>', () => {
  const props = {
    title: 'TestDocument',
    createdAt: Date.now()
  }

  test('should match its snapshot', () => {
    const component = shallow(<Document {...props}/>)
    expect(component).toMatchSnapshot()
  })

  test('should run clickHandler on button click', () => {
    const spy = jest.spyOn(Document.defaultProps, 'clickHandler')
    const component = shallow(<Document {...props}/>)
    const btn = component.find('.Document__more-btn')
    
    expect(btn.exists()).toBeTruthy()
    btn.simulate('click')
    expect(spy.mock.calls.length).toBe(1)
  })
})