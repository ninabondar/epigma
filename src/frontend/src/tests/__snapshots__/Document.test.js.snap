// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Document/> should match its snapshot 1`] = `
<div
  className="Document"
>
  <div
    className="Document__title"
  >
    TestDocument
  </div>
  <div
    className="Document__side-info"
  >
    <small>
      less than a minute
       ago
    </small>
    <div
      className="Document__more-btn"
      onClick={[Function]}
    />
  </div>
</div>
`;
